src/cli_plugin/install/kibana.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash';
+import fromRoot from '../../utils/fromRoot';
+import KbnServer from '../../server/KbnServer';
+import readYamlConfig from '../../cli/serve/read_yaml_config';
+import { statSync } from 'fs';
+
+export function existingInstall(settings, logger) {
+  try {
+    statSync(settings.plugins[0].path);
+
+    logger.error(`Plugin ${settings.plugins[0].name} already exists, please remove before installing a new version`);
+    process.exit(70); // eslint-disable-line no-process-exit
+  } catch (e) {
+    if (e.code !== 'ENOENT') throw e;
+  }
+}
+
+export async function rebuildCache(settings, logger) {
+  logger.log('Optimizing and caching browser bundles...');
+  const serverConfig = _.merge(
+    readYamlConfig(settings.config),
+    {
+      env: 'production',
+      logging: {
+        silent: settings.silent,
+        quiet: !settings.silent,
+        verbose: false
+      },
+      optimize: {
+        useBundleCache: false
+      },
+      server: {
+        autoListen: false
+      },
+      plugins: {
+        initialize: false,
+        scanDirs: [settings.pluginDir, fromRoot('src/plugins')]
+      }
+    }
+  );
+
+  const kbnServer = new KbnServer(serverConfig);
+  await kbnServer.ready();
+  await kbnServer.close();
+}
+
+export function checkVersion(settings) {
+  if (!settings.plugins[0].version) {
+    throw new Error (`Plugin version not found. Check package.json in archive`);
+  }
+
+  if (settings.plugins[0].version !== settings.version) {
```

Comment:
The version check should only be looking at the version before the `-`. For example, a plugin with the version `5.0.0-1.2.3` should install fine on Kibana `5.0.0-snapshot`